File: src/main/java/com/zr/service/CourseService.java
package com.zr.service;

import com.github.pagehelper.Page;
import com.github.pagehelper.PageHelper;
import com.zr.dao.CourseMapper;
import com.zr.pojo.Course;
import com.zr.pojo.PageResult;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Map;

@Service
public class CourseService {
    @Autowired
    CourseMapper courseMapper;

    public void deleteCourse(int[] ids){
        for (int c_id :ids){
            courseMapper.deleteCourse(c_id);
        }
    }

    public void addCourse(Course course){
        courseMapper.insertCourse(course);
    }

    public List<Course> queryAllCourse(){
        return courseMapper.queryAllCourse();
    }

    public Course findOne(int c_id){
        return courseMapper.selectCourseById(c_id);
    }

    public void updateCourse(Course course){
        courseMapper.updateCourse(course);
    }


    /**
     * 按分页查询
     */
    public PageResult findPage(int pageNum, int pageSize) {
        PageHelper.startPage(pageNum, pageSize);
        Page<Course> page= (Page<Course>) courseMapper.queryAllCourse();
        return new PageResult(page.getTotal(), page.getResult());
    }


    public List<Map> getCourseOptionList() {
       return courseMapper.selectCourseOptionList();
    }

    public int getCourseNum() {
        return courseMapper.queryAllCourse().size();
    }
}

File: src/main/webapp/js/service/courseService.js
//服务层
app.service('courseService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../course/findAll');
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../course/findPage.do?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../course/findOne?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../course/addCourse',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../course/update',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../course/deleteCourse?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.get('../course/search?page='+page+"&rows="+rows, searchEntity);
	}

	//获取课程选项
	this.getCourseOptionList=function () {
        return $http.get('../course/getCourseOptionList');
    }

});
